### PR TITLE
Add http-based vwload variant

### PR DESCRIPTION
**Pull request by wemrysi**
DRYs up the existing destinations and introduces an http-based variant utilizing `COPY VWLOAD` from a URL.

[ch11149]

### CODE REVIEW

**Comment by wemrysi** [comment]
I have verified that the azure and s3 staging variants still function properly after the refactoring. The http variant hasn't been tested, but will be as soon as we have access to an Avalanche cluster that supports it.

File: core/src/main/scala/quasar/destination/avalanche/StagedAvalancheDestination.scala [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      tableName: TableName,
+      columns: NonEmptyList[Fragment],
+      gzippedCsv: Stream[F, Byte])
+      : Stream[F, Unit] =
+    Stream.resource(stageBytes(gzippedCsv) evalMap { uri =>
+      val prepare = prepareTable(tableName, columns, writeMode, logHandler)
+      val vwload = copyVWLoad(tableName, NonEmptyList.one(uri), vwloadAuthParams, logHandler)
+
+      (prepare >> vwload).void.transact(xa)
+    })
```
Comment (rintcius) [inline core/src/main/scala/quasar/destination/avalanche/StagedAvalancheDestination.scala:65]: maybe worth deduping this too? 

```
uri =>
      val prepare = prepareTable(tableName, columns, writeMode, logHandler)
      val vwload = copyVWLoad(tableName, NonEmptyList.one(uri), vwloadAuthParams, logHandler)

      (prepare >> vwload).void.transact(xa)
```

(currently duplicated in `AvalancheHttpDestination`)

**Review by rintcius — COMMENTED** [review]
Looks great! 

I had 1 suggestion, but I'm also fine if you prefer to leave this as-is.

**Comment by wemrysi** [comment]
@rintcius ready for another look

**Review by rintcius — APPROVED** [review]

**Pull request merged**